### PR TITLE
adds rot13labs hackboard

### PR DESCRIPTION
**Pull request by c0ldbru**
## Description

Adds rot13labs hackboard as a new keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* n/a

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/rot13labs/hackboard/info.json [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        "device_version": "1.0.0"
+    },
+    "matrix_pins": {
+        "cols": ["A0", "B0", "A1", "B1", "A2", "B2", "A3", "B3", "A4", "B4", "A5", "A6", "A7", "D6", "D5", "D1", "D0", "D7"],
+        "rows": ["C2", "C3", "C7", "C4", "C6", "C5"]
+    },
+    "diode_direction": "COL2ROW",
+    "processor": "atmega32a",
+    "bootloader": "usbasploader",
+    "layouts": {
```
Comment (dunk2k) [inline keyboards/rot13labs/hackboard/info.json:18]:
```suggestion
    "community_layouts": ["tkl_ansi"],
    "layouts": {
```

File: keyboards/rot13labs/hackboard/rules.mk
Comment (dunk2k) [inline keyboards/rot13labs/hackboard/rules.mk:1]: Could the *`_ENABLE`s be migrated to `features` in `info.json`?

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/rot13labs/hackboard/rules.mk
Comment (c0ldbru) [reply keyboards/rot13labs/hackboard/rules.mk:1]: Sure thing! This has been resolved.

File: keyboards/rot13labs/hackboard/info.json (hunk 1 above)
Comment (c0ldbru) [reply keyboards/rot13labs/hackboard/info.json:27]: @dunk2k I added this change but now its throwing an error - I don't know a ton about community layouts yet but is it possible this error is being caused by the fact that the hackboard layout doesn't match the traditional tkl-ansi layout perfectly? Theres an extra key between esc and F1 on the hackboard.

Comment (c0ldbru) [reply keyboards/rot13labs/hackboard/info.json:26]: After digging into this it does appear that thats the issue - the hackboard doesnt support the tkl_ansi community layout. reverting this chnage resolves the issue.

**Comment by c0ldbru** [comment]
@dunk2k yeah, the button itself is very unique and im basically just hijacking an open spot in the grid so the button can be programmed via QMK as if it was a switch. I designed the hardware before I fully understood QMK so its not ideal, but it still works very well for my needs.

TL:DR; functionally the layout you posted is correct, even though visually the keyboard appears to match the standard community layout.

**Comment by dunk2k** [comment]
Thanks for explaining, it's that extra button is what was putting the community layout awry. Would the unique button be better placed a row, of sorts, above the Function row?

**Comment by c0ldbru** [comment]
Oh, yeah as long as its just for visuals sake I think that would be great! How/where can I do that so it wont affect the functionality?

**Comment by dunk2k** [comment]
1. `info.json`:+1 to all `"y"` co-ordinates in `layouts`, and moving the unique key to `"y": 0`. Would it be correct to assume, from the `readme.md` image, that the unique button has a red LED and his positioned top left of the micro controller?
2. *<keymap_dir>*/`keymap.c`: move the unique button's keycode position from 2nd to 1st. Could even palace it on its own row to match `layouts` in `info.json`.

File: keyboards/rot13labs/hackboard/info.json [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        "LAYOUT_tkl_ansi": {
```
Comment (dunk2k) [inline keyboards/rot13labs/hackboard/info.json:27]:
```suggestion
        "LAYOUT": {
```

File: keyboards/rot13labs/hackboard/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT_tkl_ansi(
+      KC_ESC,  KC_NO,   KC_F1,   KC_F2,   KC_F3,  KC_F4,  KC_F5,  KC_F6,  KC_F7,  KC_F8,   KC_F9,   KC_F10,   KC_F11,  KC_F12,   KC_PSCR,  KC_SCRL, KC_PAUS,
+      KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,   KC_5,   KC_6,   KC_7,   KC_8,   KC_9,    KC_0,    KC_MINS,  KC_EQL,  KC_BSPC,  KC_INS,   KC_HOME, KC_PGUP,
+      KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,   KC_O,    KC_P,    KC_LBRC,  KC_RBRC, KC_BSLS,  KC_DEL,   KC_END,  KC_PGDN,
+      KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,   KC_G,   KC_H,   KC_J,   KC_K,   KC_L,    KC_SCLN, KC_QUOT,  KC_ENT,
+      KC_LSFT,          KC_Z,    KC_X,    KC_C,   KC_V,   KC_B,   KC_N,   KC_M,   KC_COMM, KC_DOT,  KC_SLSH,  KC_RSFT,                     KC_UP,
+      KC_LCTL, KC_LGUI, KC_LALT,                          KC_SPC,                 KC_RALT, KC_RGUI, MO(1),    KC_RCTL,            KC_LEFT, KC_DOWN, KC_RIGHT),
+  
+  [1] = LAYOUT_tkl_ansi(
```
Comment (dunk2k) [inline keyboards/rot13labs/hackboard/keymaps/default/keymap.c:27]:
```suggestion
  [0] = LAYOUT(
      KC_ESC,  KC_NO,   KC_F1,   KC_F2,   KC_F3,  KC_F4,  KC_F5,  KC_F6,  KC_F7,  KC_F8,   KC_F9,   KC_F10,   KC_F11,  KC_F12,   KC_PSCR,  KC_SCRL, KC_PAUS,
      KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,   KC_5,   KC_6,   KC_7,   KC_8,   KC_9,    KC_0,    KC_MINS,  KC_EQL,  KC_BSPC,  KC_INS,   KC_HOME, KC_PGUP,
      KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,   KC_O,    KC_P,    KC_LBRC,  KC_RBRC, KC_BSLS,  KC_DEL,   KC_END,  KC_PGDN,
      KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,   KC_G,   KC_H,   KC_J,   KC_K,   KC_L,    KC_SCLN, KC_QUOT,  KC_ENT,
      KC_LSFT,          KC_Z,    KC_X,    KC_C,   KC_V,   KC_B,   KC_N,   KC_M,   KC_COMM, KC_DOT,  KC_SLSH,  KC_RSFT,                     KC_UP,
      KC_LCTL, KC_LGUI, KC_LALT,                          KC_SPC,                 KC_RALT, KC_RGUI, MO(1),    KC_RCTL,            KC_LEFT, KC_DOWN, KC_RIGHT),
  
  [1] = LAYOUT(
```

**Review by dunk2k — CHANGES_REQUESTED** [review]
conforms to PR Checklist regarding physical layout(s)

File: keyboards/rot13labs/hackboard/readme.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# HACKBOARD
+
+![hackboard](https://i.imgur.com/te8feiB.png)
```
[the remark this answers is not in the document]
Comment (c0ldbru) [reply keyboards/rot13labs/hackboard/readme.md:3]: hmmmmm, this URL is working for me? I tested it in a few browsers to be sure but its taking me to the picture as expected in each case.

**Comment by c0ldbru** [comment]
> 1. `info.json`:+1 to all `"y"` co-ordinates in `layouts`, and move the unique key to `"y": 0`. Would it be correct to assume, from the `readme.md` image, that the unique button has a red LED and his positioned top left of the micro controller on PCB?
> 2. _<keymap_dir>_/`keymap.c`: move the unique button's keycode position in sequence from 2nd to 1st. Could even place it on its own row to match `layouts` in `info.json`.

Sweet! Thanks! Thats been updated now.

File: keyboards/rot13labs/hackboard/readme.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (c0ldbru) [reply keyboards/rot13labs/hackboard/readme.md:3]: Sent this out to a few friends for independent testing - same result (image loading as expected) for each of them.

**Comment by dunk2k** [comment]
@c0ldbru 
To follow up on an earlier question: does the unique button, now on top row, have a red LED and his positioned top left of the micro controller on PCB?

**Comment by c0ldbru** [comment]
nah, its to the left of the boot and reset buttons; above F4/F5

**Comment by c0ldbru** [comment]
@dunk2k Just checking in to make sure theres nothing waiting on me for this? Let me know if theres anything I missed!

**Review by dunk2k — APPROVED** [review]

**Comment by c0ldbru** [comment]
@drashna Any chance you can review this? I believe your approval is all its waiting on.

**Comment by c0ldbru** [comment]
@dunk2k is it possible to merge this with just your approval? @drashna seems to have disappeared from this PR but I still would like to get this added for people out there using the hackboard.

**Comment by dunk2k** [comment]
> @dunk2k is it possible to merge this with just your approval? @drashna seems to have disappeared from this PR but I still would like to get this added for people out there using the hackboard.

> @c0ldbru, unfortunately I'm not a QMK maintainer, so I'm unable to merge PRs. Could I suggest on the qmk_firmware channel of the QMK Discord, relay this PR's number and ask if it could be given a look over 🙂

**Review by drashna — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**